### ISSUE TITLE
Configure custom paths to the commands used to submit jobs

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood_core#111]
You could extend this, possibly to any command used to shell out by an adapter - though it would not work for adapters that use C lib for getting job status for example.

It seems we have several use cases where being able to provide a custom wrapper around `qsub` or `sbatch`, without having to wrap the rest of the commands, would be useful. Currently you can only specify path to bin in the cluster config. The idea is you could do this:

```yaml
  job:
    adapter: "torque"
    host: "owens-batch.ten.osc.edu"
    lib: "/opt/torque/lib64"
    bin: "/opt/torque/bin"
    custom_bin:
      qsub: "/opt/torque-osc-custom/bin/qsub"
    version: "6.0.1"
```

And then when qsub is executed instead of doing `join('/opt/torque/bin', 'qsub')` to build the path to execute `qsub` is used.

Presumably this could be done in a simple way: use a hash to store these custom paths, and then have the default be a block that joins the bin path. So instead of https://github.com/OSC/ood_core/blob/602e077e761ede2cc8bc241560918423330b3db5/lib/ood_core/job/adapters/slurm.rb#L137-L144

We would do something like:

```ruby
  call("sbatch", *args, env: env, stdin: str.to_s).strip.split(";").first
end

private

def call(cmd, *args, env: {}, stdin: "")
  cmd = bin_path(cmd)
  # ... rest of method
end

def bin_path(cmd)
  custom_bin.fetch(cmd.to_sym) { bin.join(cmd.to_s).to_s }
end
```

Where `custom_bin` returns the hash form of the custom bin section, and the default value is the preexisting functionality of joining bin with the command.

We would just need to document which adapters used shell commands (and nothing else) to know what could be safely modified.

**Comment by MorganRodgers** [comment]
This breaks for things like SGE(coming soon™) (which can use DRMAA for `#info`) and for Torque which is using `libtorque.so` via the PBS gem. I am curious is Torque like SGE in that there are just huge advantages to using the FFI adapter rather than parsing command line output?

**Comment by ericfranz** [comment]
Both SGE and Torque use `qsub` so we would just have to document, per adapter, what is supported. The adapters could even have a list of legal custom_bin keys and if you try to launch with an illegal one it raises an exception (not implemented or supported or something).

**Comment by MorganRodgers** [comment]
Was merged in #112
